.github/workflows/pull-request-commentor.yaml

```diff
@@ -8,111 +8,70 @@ on:
-      pull-requests: write
-    steps:
-      - name: Add comment to trigger external storage tests for Kubernetes 1.25
-        uses: peter-evans/create-or-update-comment@v3
-        with:
-          token: ${{ secrets.CEPH_CSI_BOT_TOKEN }}
-          issue-number: ${{ github.event.pull_request.number }}
-          body: |
-            /test ci/centos/k8s-e2e-external-storage/1.25
 
-      - name: Add comment to trigger external storage tests for Kubernetes 1.26
-        uses: peter-evans/create-or-update-comment@v3
-        with:
-          token: ${{ secrets.CEPH_CSI_BOT_TOKEN }}
-          issue-number: ${{ github.event.pull_request.number }}
-          body: |
-            /test ci/centos/k8s-e2e-external-storage/1.26
-
-      - name: Add comment to trigger external storage tests for Kubernetes 1.27
-        uses: peter-evans/create-or-update-comment@v3
-        with:
-          token: ${{ secrets.CEPH_CSI_BOT_TOKEN }}
-          issue-number: ${{ github.event.pull_request.number }}
-          body: |
-            /test ci/centos/k8s-e2e-external-storage/1.27
+permissions:
+  pull-requests: write
 
-      - name: Add comment to trigger external storage tests for Kubernetes 1.28
-        uses: peter-evans/create-or-update-comment@v3
-        with:
-          token: ${{ secrets.CEPH_CSI_BOT_TOKEN }}
-          issue-number: ${{ github.event.pull_request.number }}
-          body: |
-            /test ci/centos/k8s-e2e-external-storage/1.28
+jobs:
+  branch-k8s-selection:
+    runs-on: ubuntu-latest
 
-      - name: Add comment to trigger helm E2E tests for Kubernetes 1.25
-        uses: peter-evans/create-or-update-comment@v3
-        with:
-          token: ${{ secrets.CEPH_CSI_BOT_TOKEN }}
-          issue-number: ${{ github.event.pull_request.number }}
-          body: |
-            /test ci/centos/mini-e2e-helm/k8s-1.25
+    strategy:
+      matrix:
+        branch: [release-v3.8, release-v3.9, devel]
+        k8s: ["1.25", "1.26", "1.27", "1.28"]
+        exclude:
+          # the next Ceph-CSI version will not be tested with old Kubernetes
+          - k8s: "1.25"
+            branch: devel
+          # Ceph-CSI <= 3.9 was released before Kubernetes 1.28
+          - k8s: "1.28"
+            branch: release-v3.8
+          - k8s: "1.28"
+            branch: release-v3.9
 
-      - name: Add comment to trigger helm E2E tests for Kubernetes 1.26
-        uses: peter-evans/create-or-update-comment@v3
-        with:
-          token: ${{ secrets.CEPH_CSI_BOT_TOKEN }}
-          issue-number: ${{ github.event.pull_request.number }}
-          body: |
-            /test ci/centos/mini-e2e-helm/k8s-1.26
-
-      - name: Add comment to trigger helm E2E tests for Kubernetes 1.27
-        uses: peter-evans/create-or-update-comment@v3
-        with:
-          token: ${{ secrets.CEPH_CSI_BOT_TOKEN }}
-          issue-number: ${{ github.event.pull_request.number }}
-          body: |
-            /test ci/centos/mini-e2e-helm/k8s-1.27
+    if: >
```

Comment:
@nixpanic why do we need this if check here?

Reply:
We had the if statement already, so that it filters on the `ok-to-test` label and skips PRs that are closed. The check has been extended to filter on the target branch of the PR as well.

Each job has an if-statement like this, mostly for the check on the `ok-to-test` label.